src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -492,24 +487,29 @@ public void restoreFeedbackSessionFromRecycleBin(String feedbackSessionName, Str
      * Gets the expected number of submissions for a feedback session.
      */
     public int getExpectedTotalSubmission(FeedbackSessionAttributes fsa) {
-        List<StudentAttributes> students = studentsLogic.getStudentsForCourse(fsa.getCourseId());
-        List<InstructorAttributes> instructors = instructorsLogic.getInstructorsForCourse(fsa.getCourseId());
-        List<FeedbackQuestionAttributes> questions =
-                fqLogic.getFeedbackQuestionsForSession(fsa.getFeedbackSessionName(), fsa.getCourseId());
-        List<FeedbackQuestionAttributes> studentQns = fqLogic.getFeedbackQuestionsForStudents(questions);
+        int numOfStudents = studentsLogic.getNumberOfStudentsForCourse(fsa.getCourseId());
+        List<String> instructorEmails = instructorsLogic.getInstructorEmailsForCourse(fsa.getCourseId());
 
         int expectedTotal = 0;
+        if (fqLogic.hasFeedbackQuestionsForStudents(fsa)) {
+            expectedTotal += numOfStudents;
+        }
 
-        if (!studentQns.isEmpty()) {
-            expectedTotal += students.size();
+        // Check a creator and a non-creator once each to prevent unnecessary data read.
+        List<String> creatorEmails = instructorEmails.stream()
```

Comment:
This is not needed. If the entire instructor list is iterated, this is guaranteed to be a list of size one.

Reply:
Will this guarantee hold true for the future? Then, this logic will depend on such guarantee, which doesn't seem too safe to me personally. An iteration of creator email(s) would be natural and safer should the nature of FeedbackSession creator change someday.